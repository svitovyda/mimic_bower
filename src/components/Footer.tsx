import * as React from "react";

export const Footer: React.FC = () => {
  return <div></div>;
};
