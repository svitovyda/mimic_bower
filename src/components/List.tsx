import type { Package } from '../models/Package';
import { ModuleItem } from './ModuleItem';
import * as React from 'react';

export interface ListProps {
  data: Package[];
}

export const List: React.FC<ListProps> = ({ data }: ListProps) => {
  return (
    <div>
      {data.map((p, i) => (
        <ModuleItem key={`${i}-${p.name}-${p.platform}`} packageElement={p} />
      ))}
    </div>
  );
};
