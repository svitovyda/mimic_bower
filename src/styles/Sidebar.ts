import { Button } from './ui/Button';
import styled from 'styled-components';

export const MenuButton = styled(Button)`
  width: 250px;
  margin: 0px auto 20px auto;
  display: none;

  @media only screen and (max-width: 680px) {
    display: inline-block;
  }
`;

export const SidebarContainer = styled.div`
  @media only screen and (max-width: 680px) {
    text-align: center;
    width: 100%;
  }

  @media only screen and (min-width: 681px) {
    width: 200px;
    flex-shrink: 0;
  }
`;

export const SidebarContent = styled.div<{ isShown: boolean }>`
  width: 100%;
  display: ${(props) => (props.isShown ? 'block' : 'none')};

  @media only screen and (min-width: 681px) {
    display: block;
  }
`;
