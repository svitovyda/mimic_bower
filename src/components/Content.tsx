import { useContentFetcher } from "../hooks/contentFetcher";
import { usePagination } from "../hooks/usePagination";
import type { Package } from "../models/Package";
import { List } from "./List";
import { Pagination } from "./Pagination";
import { LoaderAnimation } from "./ui/LoaderAnimation";
import configJson from "config";
import * as React from "react";

const ItemsPerPage = configJson.searchItemsPageMaxSize;

export interface ContentProps {
  query: string;
}

export const Content: React.FC<ContentProps> = ({ query }: ContentProps) => {
  const [fetchState, setQuery] = useContentFetcher("");
  const [paginationState, navigateToPage] = usePagination(fetchState.data);
  const [pageData, setPageData] = React.useState<Package[]>([]);

  React.useEffect(() => {
    setQuery(query);
    setPageData([]);
  }, [query, setQuery]);

  React.useEffect(() => {
    if (fetchState.data.length === 0) setPageData([]);
    else {
      const pageStart = (paginationState.currentPage - 1) * ItemsPerPage;

      setPageData(fetchState.data.slice(pageStart, pageStart + ItemsPerPage));
    }
  }, [fetchState.data, paginationState]);

  if (fetchState.isLoading) return <LoaderAnimation />;
  if (fetchState.isError) return <div>Sorry, there was an error while fetching repositories</div>;
  return (
    <div>
      <List data={pageData} />
      <Pagination
        currentPage={paginationState.currentPage}
        totalPages={paginationState.totalPages}
        navigateToPage={navigateToPage}
      />
    </div>
  );
};
