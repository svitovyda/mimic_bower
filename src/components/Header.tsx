import { HeaderContainer, HeaderLogo, HeaderSubtitle, HeaderTitle } from '../styles/Header';
import * as React from 'react';

export const Header: React.FC = () => {
  return (
    <header>
      <HeaderContainer>
        <HeaderLogo />
        <HeaderTitle>Bower Search</HeaderTitle>
        <HeaderSubtitle>
          Powered by{' '}
          <a href="https://libraries.io/" target="_blank" rel="noreferrer">
            libraries.io
          </a>
        </HeaderSubtitle>
      </HeaderContainer>
    </header>
  );
};
