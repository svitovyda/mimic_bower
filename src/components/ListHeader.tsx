/* eslint-disable jsx-a11y/anchor-is-valid */
import type { SortType } from '../services/ApiServise';
import { ListHeaderContainer, NameLink, OwnerLink, StarsLink } from '../styles/ListHeader';
import * as React from 'react';

export interface ListHeaderProps {
  onSort: (sortBy?: SortType) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = React.memo(({ onSort }: ListHeaderProps) => {
  return (
    <ListHeaderContainer>
      <NameLink onClick={() => onSort()}>Name</NameLink>
      <OwnerLink>Owner</OwnerLink>
      <StarsLink onClick={() => onSort('stars')}>Stars</StarsLink>
    </ListHeaderContainer>
  );
});

ListHeader.displayName = 'ListHeader';
