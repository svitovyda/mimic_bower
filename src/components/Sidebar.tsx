/* eslint-disable jsx-a11y/anchor-is-valid */
import { MenuButton, SidebarContainer, SidebarContent } from "../styles/Sidebar";
import * as React from "react";

export const Sidebar: React.FC = () => {
  const [showContent, setShowContent] = React.useState<boolean>(false);

  return (
    <SidebarContainer>
      <MenuButton onClick={() => setShowContent(!showContent)}>Menu</MenuButton>
      <SidebarContent isShown={showContent}>
        <ul>
          <li>
            <a href="#">Home</a>
          </li>
          <li>
            <a href="#">Creating Packages</a>
          </li>
          <li>
            <a href="#">API</a>
          </li>
          <li>
            <a href="#">Configuration</a>
          </li>
          <li>
            <a href="#">Pluggable Resolvers</a>
          </li>
          <li>
            <a href="#">Tools</a>
          </li>
          <li>
            <a href="#">About</a>
          </li>
        </ul>
      </SidebarContent>
    </SidebarContainer>
  );
};
