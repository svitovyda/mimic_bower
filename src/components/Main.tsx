import { MainContainer, RightContainer } from "../styles/Main";
import { Content } from "./Content";
import { Footer } from "./Footer";
import { Search } from "./Search";
import { Sidebar } from "./Sidebar";
import * as React from "react";

export const Main: React.FC = () => {
  const [query, setQuery] = React.useState<string>("");

  const onNewQuery = React.useCallback(
    (newQuery: string) => {
      setQuery(newQuery);
    },
    [setQuery]
  );

  return (
    <MainContainer>
      <Sidebar />
      <RightContainer>
        <Search onQueryChanged={onNewQuery} />
        <Content query={query} />
        <Footer />
      </RightContainer>
    </MainContainer>
  );
};
