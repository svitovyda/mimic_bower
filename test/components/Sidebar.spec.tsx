import { Sidebar } from "../../src/components/Sidebar";
import { render } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";

describe("Sidebar", () => {
  it("renders without error", () => {
    const component = render(<Sidebar />);

    expect(component).toMatchSnapshot();
  });
});
