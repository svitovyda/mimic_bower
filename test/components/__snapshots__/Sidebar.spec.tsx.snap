// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

.c2 {
  width: 250px;
  margin: 0px auto 20px auto;
  display: none;
}

.c3 {
  width: 100%;
  display: none;
}

@media only screen and (max-width:680px) {
  .c2 {
    display: inline-block;
  }
}

@media only screen and (max-width:680px) {
  .c0 {
    text-align: center;
    width: 100%;
  }
}

@media only screen and (min-width:681px) {
  .c0 {
    width: 200px;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

@media only screen and (min-width:681px) {
  .c3 {
    display: block;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <button
          class="c1 c2"
        >
          Menu
        </button>
        <div
          class="c3"
        >
          <ul>
            <li>
              <a
                href="#"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Creating Packages
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                API
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Configuration
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Pluggable Resolvers
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Tools
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                About
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c1:hover:enabled {
  border-color: #ef5734;
}

.c1:disabled {
  color: lightgray;
  border-color: lightgray;
}

.c2 {
  width: 250px;
  margin: 0px auto 20px auto;
  display: none;
}

.c3 {
  width: 100%;
  display: none;
}

@media only screen and (max-width:680px) {
  .c2 {
    display: inline-block;
  }
}

@media only screen and (max-width:680px) {
  .c0 {
    text-align: center;
    width: 100%;
  }
}

@media only screen and (min-width:681px) {
  .c0 {
    width: 200px;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

@media only screen and (min-width:681px) {
  .c3 {
    display: block;
  }
}

<div>
    <div
      class="c0"
    >
      <button
        class="c1 c2"
      >
        Menu
      </button>
      <div
        class="c3"
      >
        <ul>
          <li>
            <a
              href="#"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Creating Packages
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              API
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Configuration
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Pluggable Resolvers
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Tools
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
